### ISSUE TITLE
Compound errors are swallowed in LoggingContextJob

### ISSUE THREAD

**Issue by philippthun** [cloudfoundry/cloud_controller_ng#2147]
## Issue

Error information for compound errors (V3) is hidden behind a generic `UnknownError` in `LoggingContextJob`.

## Context

The deletion of an app failed with the following output:
```
$ cf d myapp
Really delete the app myapp? [yN]: y
Deleting app myapp in org myorg / space myspace as myuser...
Job (4183619e-ac35-4ff6-adb4-e6e2504c07e1) failed: An unknown error occurred.
FAILED
```

The log message contained more information (see `test_mode_info`, especially `CF-UnprocessableDelete`):
```
{"message":"Request failed: 500: {\"error_code\"=>\"UnknownError\", \"description\"=>\"An unknown error occurred.\", \"code\"=>10001, \"test_mode_info\"=>{\"description\"=>\"Service broker failed to delete service binding for instance myservice: Service broker error: Service broker mybroker responded with invalid JSON: 404 Not Found: Requested route ('mybroker.myfoundry.com') does not exist.\\n\", \"error_code\"=>\"CF-UnprocessableDelete\", \"backtrace\"=...
```

We also had a look into the Cloud Controller database (table `delayed_jobs`); the `last_error` field contained detailed information (the service broker error), but field `cf_api_error` showed `UnknownError`.

When searching through the code, we found out that when the service binding deletion failed, a `V3::ServiceBindingDelete::UnprocessableDelete` error was wrapped into an `AppDelete::SubResourceError`. This error was then translated (`DeleteAppErrorTranslatorJob`) into a `CloudController::Errors::ApiError` wrapped into a `CloudController::Errors::CompoundError`, but the `LoggingContextJob` (https://github.com/cloudfoundry/cloud_controller_ng/blob/c0b9a4ae88e169e9b801377feebc5bfb5c00c3a5/app/jobs/logging_context_job.rb#L24) does only use the `V2ErrorHasher` which is - in contrast to the `V3ErrorHasher` not aware of compound errors and thus renders the exception into an `UnknownError`.

## Expected result

The cf cli should show the underlying error, e.g. the description of the `CF-UnprocessableDelete` error.

## Possible Fix

Utilize `V3ErrorHasher` in `LoggingContextJob` (for compound errors).

**Comment by JenGoldstrich** [comment]
Hey @philippthun Thank you so much for reporting this, once we see your[ PR ](https://github.com/cloudfoundry/cloud_controller_ng/pull/2148) goes through the pipeline I will close this issue out.

Thanks!
Jenna

**Comment by JenGoldstrich** [comment]
Hey @philippthun your change made it through our pipeline successfully, I am adding a note to our release task that once we release your item we will close this issue, I want to leave this open for the mean time to capture this issue in the current version of capi-release

**Comment by sweinstein22** [comment]
We've released [1.108.0](https://github.com/cloudfoundry/capi-release/releases/tag/1.108.0), which includes the PR associated with this issue!
